provider/kubeletconfig/kubeletconfig_resource.go

```diff
@@ -0,0 +1,261 @@
+	if err != nil || exists {
+		resp.Diagnostics.AddError(failedToCreateSummary,
+			fmt.Sprintf("KubeletConfig for cluster '%s' may already exist: %v", clusterId, err))
+		return
+	}
+
+	kubeletConfig, err := k.convertStateToApiResource(state)
+	if err != nil {
+		resp.Diagnostics.AddError(failedToCreateSummary,
+			fmt.Sprintf("Failed to build request to create KubeletConfig on cluster '%s': %v", clusterId, err))
+		return
+	}
+
+	_, err = k.configClient.Create(ctx, clusterId, kubeletConfig)
+	if err != nil {
+		resp.Diagnostics.AddError(failedToCreateSummary,
+			fmt.Sprintf("Failed to create KubeletConfig on cluster '%s': %v", clusterId, err))
+		return
+	}
+
+	k.writeStateToResponse(ctx, state, &resp.State, &resp.Diagnostics)
+}
+
+func (k *KubeletConfigResource) convertStateToApiResource(state *KubeletConfigState) (*cmv1.KubeletConfig, error) {
+	builder := cmv1.KubeletConfigBuilder{}
+	builder.PodPidsLimit(int(state.PodPidsLimit.ValueInt64()))
+	return builder.Build()
+}
+
+func (k *KubeletConfigResource) convertApiResourceToState(
+	kubeletConfig *cmv1.KubeletConfig, state *KubeletConfigState) {
+	state.PodPidsLimit = types.Int64Value(int64(kubeletConfig.PodPidsLimit()))
+}
+
+func (k *KubeletConfigResource) getKubeletConfigStateFromState(
+	ctx context.Context, state tfsdk.State, diags *diag.Diagnostics) *KubeletConfigState {
+	ks := &KubeletConfigState{}
+	ds := state.Get(ctx, ks)
+	diags.Append(ds...)
+	return ks
+}
+
+func (k *KubeletConfigResource) getKubeletConfigStateFromPlan(
+	ctx context.Context, plan tfsdk.Plan, diags *diag.Diagnostics) *KubeletConfigState {
+	ks := &KubeletConfigState{}
+	ds := plan.Get(ctx, ks)
+	diags.Append(ds...)
+	return ks
+}
+
+func (k *KubeletConfigResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	state := k.getKubeletConfigStateFromState(ctx, req.State, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	clusterId := state.Cluster.ValueString()
+	exists, kubeletConfig, err := k.configClient.Exists(ctx, clusterId)
+	if err != nil {
+		resp.Diagnostics.AddError("Cannot read KubeletConfig",
+			fmt.Sprintf("Cannot read KubeletConfig for cluster '%s': %v", clusterId, err))
+		return
+	}
+
+	if !exists {
+		tflog.Warn(ctx, fmt.Sprintf("kubeletconfig for cluster (%s) not found, removing from state",
+			state.Cluster.ValueString(),
+		))
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	k.convertApiResourceToState(kubeletConfig, state)
+	k.writeStateToResponse(ctx, state, &resp.State, &resp.Diagnostics)
+}
+
+func (k *KubeletConfigResource) writeStateToResponse(
+	ctx context.Context, ks *KubeletConfigState, tfState *tfsdk.State, diagnostics *diag.Diagnostics) {
+
+	diags := tfState.Set(ctx, ks)
+	diagnostics.Append(diags...)
+}
+
+func (k *KubeletConfigResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+
+	state := k.getKubeletConfigStateFromPlan(ctx, req.Plan, &resp.Diagnostics)
```

Comment:
I suggest to use here "plan" and not "state" to make it clear that the plan is used here (you can get the state as well)
```suggestion
	plan := k.getKubeletConfigState(ctx, req.Plan, &resp.Diagnostics)
```